### DISCUSSION TITLE
Unable to resolve "defineNuxtRouterMiddleware" definition

### DISCUSSION THREAD

**Discussion by Amphitrite632** [nuxt/nuxt#19322] [Questions]
I tried to do a redirect using middleware with TypeScript, but I get an error `Cannot find name 'defineNuxtRouteMiddleware'`. It seems that TSC is unable to resolve the Nuxt libraries.
For now this error can be ignored as no transpile is required to run the middleware, but for the sake of mental health I would like to resolve this.

Here is my `package.json` :
```json
"devDependencies": {
        "@nuxt/eslint-config": "^0.1.1",
        "@nuxt/types": "^2.16.0",
        "@types/node": "^18.14.1",
        "nuxt": "^3.2.2",
        "typescript": "^4.9.5"
}
```
Perhaps I'm missing something here instead of running into a bug or something.

Thanks. (And sorry for my poor English)

**Comment by AloisSeckar** [comment]
There might be few things:

1. Are you sure you have `defineNuxtRouteMiddleware`? Because in topic title you wrote "defineNuxtRoute**r**Middleware" (with an extra "r")
2. For me Nuxt build-ins don't get resolved well even after `npm install` before the first time when `npm run dev` runs, where I believe some linking `.d.ts` files are being created.
3. Do you use VS Code and have [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) with [Takeover mode](https://vuejs.org/guide/typescript/overview.html#volar-takeover-mode) up and running?
